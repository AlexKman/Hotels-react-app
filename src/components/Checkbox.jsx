import React from "react";

const Checkbox = props => {
  return <input type="checkbox" onClick={props.onClick} id={props.id} />;
};

export default Checkbox;
