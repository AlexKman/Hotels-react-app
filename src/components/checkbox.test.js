import { mount } from "enzyme";
import React from "react";
import Checkbox from "./checkbox";

describe("checkbox", () => {
  let component;
  let mockFn;
  let id;
  beforeEach(() => {
    mockFn = jest.fn();
    id = "123";
    component = mount(<Checkbox id={id} onClick={mockFn} />);
  });
  it("renders correctly", () => {
    expect(component).toMatchSnapshot();
  });
  it("executes onClick handler on click", () => {
    component.find("input").simulate("click");
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
  it("has id as given by the prop", () => {
    expect(component.find({ id }).exists()).toEqual(true);
  });
});
