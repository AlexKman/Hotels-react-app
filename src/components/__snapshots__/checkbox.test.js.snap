// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox renders correctly 1`] = `
<Checkbox
  id="123"
  onClick={[MockFunction]}
>
  <input
    id="123"
    onClick={[MockFunction]}
    type="checkbox"
  />
</Checkbox>
`;
